Applications/Spire/Include/Spire/Ui/DropDownBox.hpp

```diff
@@ -138,24 +140,33 @@ namespace Styles {
     protected:
       bool eventFilter(QObject* watched, QEvent* event) override;
       void keyPressEvent(QKeyEvent* event) override;
+      void mousePressEvent(QMouseEvent* event) override;
 
     private:
       mutable SubmitSignal m_submit_signal;
       ListView* m_list_view;
       TextBox* m_text_box;
       Button* m_button;
       DropDownList* m_drop_down_list;
+      QTimer m_timer;
       bool m_is_read_only;
       boost::optional<int> m_submission;
+      boost::optional<PressObserver> m_button_press_observer;
       bool m_is_modified;
+      bool m_is_mouse_press_on_list;
+      QPoint m_mouse_press_position;
+      QPointer<QWidget> m_hovered_item;
       boost::signals2::scoped_connection m_submit_connection;
       boost::signals2::scoped_connection m_current_connection;
 
-      void on_click();
-      void on_current(const boost::optional<int>& current);
-      void on_submit(const std::any& submission);
+      int get_index_under_mouse(const QPoint& global_point) const;
+      void hover_enter(const QMouseEvent& event);
```

Comment:
Function names need to be verbs. You can rename this to ```on_hover_enter``` / ```on_hover_leave```, or you can rename it to the actual function/verb that is being performed.